### PR TITLE
Db 1692 d32-d34 period of performance dates

### PR DESCRIPTION
**Pull request by alburde1**
Creates rules D32-D34 pertaining to PeriodOfPerformanceStartDate and PeriodOfPerformanceCurrentEndDate.

Technical Release Notes:
- New sql rules

### CODE REVIEW

File: dataactvalidator/config/sqlrules/d32_detached_award_financial_assistance_1.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  return FALSE;
+END;
+$$ LANGUAGE plpgsql;
+
+SELECT
+    row_number,
+    period_of_performance_star
+FROM detached_award_financial_assistance
+WHERE submission_id = {0}
+    AND period_of_performance_star IS NOT NULL
```
Comment (nmonga91) [inline dataactvalidator/config/sqlrules/d32_detached_award_financial_assistance_1.sql]: do we want this to fail if it's blank? or should we treat blank and NULL the same?

File: dataactvalidator/config/sqlrules/d33_detached_award_financial_assistance_1.sql [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  return FALSE;
+END;
+$$ LANGUAGE plpgsql;
+
+SELECT
+    row_number,
+    period_of_performance_curr
+FROM detached_award_financial_assistance
+WHERE submission_id = {0}
+    AND period_of_performance_curr IS NOT NULL
```
Comment (nmonga91) [inline dataactvalidator/config/sqlrules/d33_detached_award_financial_assistance_1.sql]: Same as the previous question for blanks vs nulls

File: tests/unit/dataactvalidator/test_d32_detached_award_financial_assistance_1.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+def test_column_headers(database):
+    expected_subset = {"row_number", "period_of_performance_star"}
+    actual = set(query_columns(_FILE, database))
+    assert expected_subset == actual
+
+
+def test_success(database):
+    """ PeriodOfPerformanceStartDate is an optional field, but when provided, must follow YYYYMMDD format """
+    det_award_1 = DetachedAwardFinancialAssistanceFactory(period_of_performance_star="19990131")
```
Comment (nmonga91) [inline tests/unit/dataactvalidator/test_d32_detached_award_financial_assistance_1.py]: we should probably add a test case for a null value to make sure it's omitted. And if we decide to avoid blanks, a blank test case should also be added.

File: tests/unit/dataactvalidator/test_d33_detached_award_financial_assistance_1.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+def test_column_headers(database):
+    expected_subset = {"row_number", "period_of_performance_curr"}
+    actual = set(query_columns(_FILE, database))
+    assert expected_subset == actual
+
+
+def test_success(database):
+    """ PeriodOfPerformanceCurrentEndDate is an optional field, but when provided, must follow YYYYMMDD format """
+    det_award_1 = DetachedAwardFinancialAssistanceFactory(period_of_performance_curr="19990131")
```
Comment (nmonga91) [inline tests/unit/dataactvalidator/test_d33_detached_award_financial_assistance_1.py]: Same comment as D32_1 for nulls/blanks

File: tests/unit/dataactvalidator/test_d33_detached_award_financial_assistance_2.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    expected_subset = {"row_number", "period_of_performance_curr"}
+    actual = set(query_columns(_FILE, database))
+    assert expected_subset == actual
+
+
+def test_success(database):
+    """ When provided, PeriodOfPerformanceCurrentEndDate must be a valid date between 19991001 and 20991231.
+        (i.e., a date between 10/01/1999 and 12/31/2099) """
+    det_award_1 = DetachedAwardFinancialAssistanceFactory(period_of_performance_curr="20120725")
+    det_award_2 = DetachedAwardFinancialAssistanceFactory(period_of_performance_curr=None)
```
Comment (nmonga91) [inline tests/unit/dataactvalidator/test_d33_detached_award_financial_assistance_2.py]: We should also probably add a test case for blank here too, as well as other tests

File: dataactvalidator/config/sqlrules/d32_detached_award_financial_assistance_1.sql (hunk 1 above)
Comment (alburde1) [inline dataactvalidator/config/sqlrules/d32_detached_award_financial_assistance_1.sql]: I did this a while ago but I think I tested and if someone enters blank or just spaces it defaults to null in the DB

Comment (nmonga91) [inline dataactvalidator/config/sqlrules/d32_detached_award_financial_assistance_1.sql]: Ok cool. If the broker inserts it properly when uploaded we may be good. But I know that for the unit tests, we can force a blank value so not sure how we want to handle that in the SQL

Comment (alburde1) [inline dataactvalidator/config/sqlrules/d32_detached_award_financial_assistance_1.sql]: I can just put it in. I was just trying to simplify but it certainly won't hurt anything.

**Comment by nmonga91** [comment]
Looks good, ready to merge once checks have completed.
